storey/flow.py

```diff
@@ -1769,7 +1773,14 @@ def __init__(
         if not runnables:
             raise ValueError("ParallelExecution cannot be instantiated without at least one runnable")
 
+        for runnable in runnables:
+            execution_mechanism = execution_mechanism_by_runnable_name.get(runnable.name)
+            if not execution_mechanism:
+                raise ValueError(f"No execution mechanism was specified for runnable '{runnable.name}'")
```

Comment:
```suggestion
            if not execution_mechanism := execution_mechanism_by_runnable_name.get(runnable.name):
                raise ValueError(f"No execution mechanism was specified for runnable '{runnable.name}'")
```

Reply:
Maybe you can delete it and to do this validation on `add_runnable`

Reply:
We could, but it would be less informative without the runnable name and without the ability to distinguish `None` from it not being provided in the first place.

Reply:
OK so lets leave it.. can you edit it?